pkg/webhook/webhook.go

```diff
@@ -55,15 +59,49 @@ func (q *QuayIntegrationMutator) Handle(ctx context.Context, req admission.Reque
 			}
 		}
 	} else {
-
-		admissionResponse = getAdmissionResponseForBuild(build, &quayIntegration)
+		// Check if builder service account has secret
+		hasSecret, serviceAcctErr := q.checkSecretForBuilderServiceAccount(ctx, &req, &quayIntegration)
+
+		if !hasSecret {
+			if serviceAcctErr != nil {
+				admissionResponse = &admissionv1.AdmissionResponse{
+					Allowed: false,
+					Result: &metav1.Status{
+						Message: serviceAcctErr.Error(),
+					},
+				}
+			} else {
+				admissionResponse = &admissionv1.AdmissionResponse{
+					Allowed: false,
```

Comment:
It's better to include a message like "the builder service account has not been provisioned with secrets yet"

Reply:
Agree. There has to be a friendly note for either error condition

Reply:
added a msg